src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -113,29 +114,56 @@ def register_options(cls, register):
             # TODO(#11719): Switch this to a file_option once env vars can unset a value.
             type=str,
             metavar="<file>",
-            mutually_exclusive_group="constraints",
+            mutually_exclusive_group="lockfile",
             help=(
                 "The lockfile to use when resolving requirements for your own code (vs. tools you "
                 "run).\n\n"
-                "This is highly experimental and will change, including adding support for "
-                "multiple lockfiles. This option's behavior may change without the normal "
-                "deprecation cycle.\n\n"
+                "This is highly experimental and will be replaced by `[python].enable_resolves`.\n\n"
                 "To generate a lockfile, activate the backend `pants.backend.experimental.python`"
                 "and run `./pants generate-user-lockfile ::`.\n\n"
-                "Mutually exclusive with `[python].requirement_constraints`."
+                "Mutually exclusive with `[python].requirement_constraints` and "
+                "`[python].enable_resolves`."
+            ),
+        )
+        register(
+            "--enable-resolves",
```

Comment:
Would it be ambiguous to call this "require-lockfile" or something? It's not quite clear what "enabling resolves" would mean, and most users will only be enabling lockfiles, rather than enabling multiple resolves.

Reply:
I think it'd be a little confusing because we've in the past called "constraints" a "lockfile", so it's an overloaded term. And we have `--experimental-lockfile` which we know a few users are using.

I went with `--enable-resolves` because the overall new feature can be called "resolves" to differentiate from the status quo. Even if you only have a single resolve, you still have fields like `compatible_resolves`, `resolve`, and `--default-resolve`. "resolve" is the main way we refer to this feature.